init.lua

```diff
@@ -30,6 +30,20 @@ local function destroy(handle)
 end
 ffi.gc(cudnn.handle, destroy)
 
+cudnn.typemap = {
+   ['torch.CudaHalfTensor']   = 'CUDNN_DATA_HALF',
+   ['torch.CudaTensor']       = 'CUDNN_DATA_FLOAT',
+   ['torch.CudaDoubleTensor'] = 'CUDNN_DATA_DOUBLE',
+}
+
+-- TODO: determine if device supports true half and use true half on it
```

Comment:
Please disregard this one, I had cmeke issue when building it - arch settings were not right.
Most oftyhe test run fine on both 5.x and 6.0 now, except this one for functional.lua:

functional_convolution2d
 Function call failed
/usr/share/lua/5.1/cudnn/init.lua:58: Error in CuDNN: CUDNN_STATUS_INVALID_VALUE (cudnnConvolutionForward)
stack traceback:
        [C]: in function 'error'
        /usr/share/lua/5.1/cudnn/init.lua:58: in function 'errcheck'
        /usr/share/lua/5.1/cudnn/functional.lua:117: in function 'Convolution2D_updateOutput'
        extra/cudnn/test/test.lua:1413: in function <extra/cudnn/test/test.lua:1403>
        [C]: in function 'xpcall'
        /usr/share/lua/5.1/torch/Tester.lua:477: in function '_pcall'
        /usr/share/lua/5.1/torch/Tester.lua:436: in function '_run'
        /usr/share/lua/5.1/torch/Tester.lua:355: in function 'run'
        extra/cudnn/test/test.lua:1469: in main chunk
        [C]: in function 'dofile'
        /usr/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
        [C]: at 0x00406670

--------------------------------------------------------------------------------
/usr/bin/luajit: /usr/share/lua/5.1/torch/Tester.lua:363: An error was found while running tests!
stack traceback:
        [C]: in function 'assert'
        /usr/share/lua/5.1/torch/Tester.lua:363: in function 'run'
        extra/cudnn/test/test.lua:1469: in main chunk
        [C]: in function 'dofile'
        /usr/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
        [C]: at 0x00406670